test/mysql.test.js

```diff
@@ -424,6 +431,101 @@ describe('mysql', function() {
+      PostWithNumId.destroyAll(done);
+    });
+    after(function(done) {
+      PostWithNumId.destroyAll(done);
+    });
+    it('should handle null in inq operator', function(done) {
+      defaultPost.id = 1;
+      defaultPost.title = 'Foo';
+      defaultPost.content = 'Bar';
+      PostWithNumId.create(defaultPost, function(err, post) {
+        should.not.exist(err);
+        post.id.should.equal(defaultPost.id);
+        PostWithNumId.find({where: {id: {inq: [null, 1]}}}, function(err, posts) {
+          should.not.exist(err);
+          posts.length.should.equal(1);
+          posts[0].title.should.equal('Foo');
+          posts[0].id.should.equal(1);
+          done();
+        });
+      });
+    });
+
+    it('should handle null in nin operator', function(done) {
+      defaultPost.id = 2;
+      defaultPost.title = 'Make';
+      defaultPost.content = 'Toyota';
+      PostWithNumId.create(defaultPost, function(err, post) {
+        should.not.exist(err);
+        post.id.should.equal(defaultPost.id);
+        PostWithNumId.find({where: {id: {nin: [null, 3]}}}, function(err, posts) {
+          should.not.exist(err);
+          posts.length.should.equal(1);
+          posts[0].content.should.equal('Toyota');
+          posts[0].id.should.equal(2);
+          done();
+        });
+      });
+    });
+
+    it('should handle null in neq operator', function(done) {
+      defaultPost.id = 3;
+      defaultPost.title = 'Model';
+      defaultPost.content = 'Corolla';
+      PostWithNumId.create(defaultPost, function(err, post) {
+        should.not.exist(err);
+        post.id.should.equal(defaultPost.id);
+        PostWithNumId.find({where: {id: {neq: null}}}, function(err, posts) {
+          should.not.exist(err);
+          posts.length.should.equal(1);
+          posts[0].content.should.equal('Corolla');
+          posts[0].id.should.equal(3);
+          done();
+        });
+      });
+    });
+
+    it('should handle null in nin op for different datatypes', function(done) {
+      PostWithNumId.create(defaultPost, function(err, post) {
+        should.not.exist(err);
+        post.id.should.equal(defaultPost.id);
+        PostWithNumId.find({where: {and: [
+        {id: {nin: [null]}},
+        {title: {nin: [null]}},
+        {content: {nin: [null]}},
+        {buffProp: {nin: [null]}},
+        {objProp: {nin: [null]}},
+        {arrProp: {nin: [null]}},
+        {dateProp: {nin: [null]}},
+        {pointProp: {nin: [null]}},
+        ]}}, function(err, posts) {
+          should.not.exist(err);
+          posts.length.should.equal(1);
+          posts[0].id.should.equal(defaultPost.id);
+          posts[0].title.should.equal(defaultPost.title);
+          posts[0].content.should.equal(defaultPost.content);
+          posts[0].buffProp.should.deepEqual(defaultPost.buffProp);
+          posts[0].objProp.should.deepEqual(defaultPost.objProp);
+          posts[0].pointProp.should.deepEqual(defaultPost.pointProp);
```

Comment:
If you're concerned the meaning of the test will not be easily conveyed to the reader, then you could loop through the properties and assert equality programmatically, since both objects should share the same property keys.
```
for (var key in defaultPost) {
  posts[0][key].should.equal(defaultPost[key]);
}
```